components/brave_news/browser/brave_news_controller.cc

```diff
@@ -891,18 +918,43 @@ void BraveNewsController::OnChannelsChanged() {
                                  weak_ptr_factory_.GetWeakPtr())));
 
     // When channels are changed, see if it affects the feed.
-    if (MaybeInitFeedV2()) {
-      feed_v2_builder_->RecheckFeedHash(last_subscriptions_);
-    }
+    // TODO: We should fire a callback if an update is available, and notify
+    // listeners.
+    IN_ENGINE(CheckForFeedsUpdate,
+              base::BindOnce(&BraveNewsController::NotifyFeedChanged,
+                             weak_ptr_factory_.GetWeakPtr()),
+              /*refetch_data=*/false);
   } else {
     VLOG(1) << "OnChannelsChanged: News not enabled, doing nothing.";
   }
 }
 
 void BraveNewsController::NotifyChannelsChanged(mojom::ChannelsEventPtr event) {
+  DVLOG(1) << __FUNCTION__;
   for (const auto& observer : channels_listeners_) {
     observer->Changed(event->Clone());
   }
 }
 
+void BraveNewsController::NotifyFeedChanged(const std::string& hash) {
+  DVLOG(1) << __FUNCTION__;
+  for (const auto& observer : feed_listeners_) {
+    observer->OnUpdateAvailable(hash);
+  }
+}
+
+BackgroundHistoryQuerier BraveNewsController::MakeHistoryQuerier() {
+  return brave_news::MakeHistoryQuerier(
+      history_service_->AsWeakPtr(),
+      base::BindRepeating(
+          [](base::WeakPtr<BraveNewsController> controller)
+              -> base::CancelableTaskTracker* {
+            return controller ? &controller->task_tracker_ : nullptr;
+          },
+          weak_ptr_factory_.GetWeakPtr()));
+}
+
+#undef IN_ENGINE
+#undef IN_ENGINE_FF
```

Comment:
nit: maybe undefying last one first?
```suggestion
#undef IN_ENGINE_FF
#undef IN_ENGINE
```